servers/zts/src/main/java/com/yahoo/athenz/zts/cert/impl/DynamoDBCertRecordStoreConnection.java

```diff
@@ -310,23 +315,50 @@ private boolean isRecordUpdatedWithNotificationTimeAndServer(String lastNotified
         List<String> unrefreshedCertDates = DynamoDBUtils.getISODatesByRange(unrefreshedCertsRangeBegin, unrefreshedCertsRangeEnd);
 
         for (String unrefreshedCertDate : unrefreshedCertDates) {
-            items.addAll(getUnrefreshedCertRecordsByDate(provider, index, yesterday, unrefreshedCertDate));
+            items.addAll(getUnrefreshedCertRecordsByDate(provider, yesterday, unrefreshedCertDate));
         }
 
+        // Filter outdated records from before re-bootstrapping (another record exist with a new uuid)
+        items = items.stream()
+                .filter(item -> (mostUpdatedHostRecord(item)))
+                .collect(Collectors.toList());
+
         return items;
     }
 
-    private List<Item> getUnrefreshedCertRecordsByDate(String provider, Index index, long yesterday, String unrefreshedCertDate) {
+    private boolean mostUpdatedHostRecord(Item recordToCheck) {
+        try {
+            // Query all records with the same hostName as recordToCheck
+            QuerySpec spec = new QuerySpec()
```

Comment:
For every record:
1. Get all records with the same hostName
2. Check which one is the most up to date (biggest currentTime).
3. Check if it is the same record we evaluate now. If it is we will send a notification if not we'll skip it.